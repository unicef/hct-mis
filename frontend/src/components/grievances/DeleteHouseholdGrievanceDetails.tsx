import { Box, Button, Grid, Typography } from '@material-ui/core';
import { Field, Formik } from 'formik';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { useSnackbar } from '../../hooks/useSnackBar';
import { FormikRadioGroup } from '../../shared/Formik/FormikRadioGroup';
import { FormikTextField } from '../../shared/Formik/FormikTextField';
import { GRIEVANCE_TICKET_STATES } from '../../utils/constants';
import { choicesToDict } from '../../utils/utils';
import {
  GrievanceTicketDocument,
  GrievanceTicketQuery,
  useApproveDeleteHouseholdDataChangeMutation,
  useHouseholdChoiceDataQuery,
} from '../../__generated__/graphql';
import { useConfirmation } from '../core/ConfirmationDialog';
import { ContentLink } from '../core/ContentLink';
import { LabelizedField } from '../core/LabelizedField';
import { LoadingComponent } from '../core/LoadingComponent';
import { Title } from '../core/Title';
import { ApproveBox } from './GrievancesApproveSection/ApproveSectionStyles';

export const DeleteHouseholdGrievanceDetails = ({
  ticket,
  canApproveDataChange,
}: {
  ticket: GrievanceTicketQuery['grievanceTicket'];
  canApproveDataChange: boolean;
}): React.ReactElement => {
  const { t } = useTranslation();
  const { showMessage } = useSnackbar();
  const businessArea = useBusinessArea();
  const [mutate] = useApproveDeleteHouseholdDataChangeMutation();
  const confirm = useConfirmation();
  const {
    data: choicesData,
    loading: choicesLoading,
  } = useHouseholdChoiceDataQuery();

  const isForApproval = ticket.status === GRIEVANCE_TICKET_STATES.FOR_APPROVAL;
  const approveEnabled = isForApproval;
  const dialogText = t(
    'You did not approve the following household to be withdrawn. Are you sure you want to continue?',
  );

  const dialogContentWithdrawReason = (
    <Formik
      initialValues={{ withdrawReason: '', duplicateHouseholdId: null }}
      enableReinitialize
      onSubmit={(values) => {
        console.log('VALUES', values);
      }}
    >
      {({ submitForm }) => (
        <Box display='flex' flexDirection='column'>
          <Box mt={2}>
            <Typography variant='body2'>
              {t('Please provide the reason of withdrawal of this household.')}
            </Typography>
          </Box>
          <Field
            name='withdrawReason'
            choices={[
              {
                value: 'duplicate',
                name: 'This household is a duplicate of another household',
              },
            ]}
            component={FormikRadioGroup}
            noMargin
          />
          <Grid container>
            <Grid item xs={6}>
              <Field
                name='duplicateHouseholdId'
                fullWidth
                variant='outlined'
                label={t('Household Id')}
                component={FormikTextField}
                required
              />
            </Grid>
          </Grid>
          <Field
            name='withdrawReason'
            choices={[{ value: 'other', name: 'Other' }]}
            component={FormikRadioGroup}
            noMargin
          />
          <Button onClick={submitForm}>Submit</Button>
        </Box>
      )}
    </Formik>
  );

  const { approveStatus } = ticket.deleteHouseholdTicketDetails;

  if (choicesLoading) return <LoadingComponent />;

  const residenceChoicesDict = choicesToDict(
    choicesData.residenceStatusChoices,
  );

  return (
    <ApproveBox>
      <Title>
        <Box display='flex' justifyContent='space-between'>
          <Typography variant='h6'>{t('Household to be withdrawn')}</Typography>
          {canApproveDataChange && (
            <Button
              onClick={() =>
                confirm({
                  title: approveStatus
                    ? t('Warning')
                    : t('Reason of withdrawal'),
                  content: approveStatus
                    ? dialogText
                    : dialogContentWithdrawReason,
                  continueText: t('Confirm'),
                }).then(async () => {
                  try {
                    await mutate({
                      variables: {
                        grievanceTicketId: ticket.id,
                        approveStatus: !ticket.deleteHouseholdTicketDetails
                          ?.approveStatus,
                      },
                      refetchQueries: () => [
                        {
                          query: GrievanceTicketDocument,
                          variables: { id: ticket.id },
                        },
                      ],
                    });
                    if (approveStatus) {
                      showMessage(t('Changes Disapproved'));
                    }
                    if (!approveStatus) {
                      showMessage(t('Changes Approved'));
                    }
                  } catch (e) {
                    e.graphQLErrors.map((x) => showMessage(x.message));
                  }
                })
              }
              variant={approveStatus ? 'outlined' : 'contained'}
              color='primary'
              disabled={!approveEnabled}
            >
              {approveStatus ? t('Disapprove') : t('Approve')}
            </Button>
          )}
        </Box>
      </Title>
      <Grid container spacing={6}>
        <Grid item xs={3}>
          <LabelizedField label={t('Household Size')}>
            {ticket.household.size}
          </LabelizedField>
        </Grid>
        <Grid item xs={3}>
          <LabelizedField label={t('Residence Status')}>
            {residenceChoicesDict[ticket.household.residenceStatus]}
          </LabelizedField>
        </Grid>
        <Grid item xs={3}>
          <LabelizedField label={t('Head of Household')}>
            <ContentLink
              href={`/${businessArea}/population/individuals/${ticket.household.headOfHousehold.id}`}
            >
              {ticket.household.headOfHousehold.fullName}
            </ContentLink>
          </LabelizedField>
        </Grid>
        <Grid item xs={3}>
          <LabelizedField label={t('Country')}>
            {ticket.household.country}
          </LabelizedField>
        </Grid>

        <Grid item xs={3}>
          <LabelizedField label={t('Country of Origin')}>
            {ticket.household.countryOrigin}
          </LabelizedField>
        </Grid>
        <Grid item xs={3}>
          <LabelizedField label={t('Address')}>
            {ticket.household.address}
          </LabelizedField>
        </Grid>
        <Grid item xs={3}>
          <LabelizedField label={t('Village')}>
            {ticket.household.village}
          </LabelizedField>
        </Grid>
        <Grid item xs={3}>
          <LabelizedField label={t('Administrative Level 1')}>
            {ticket.household.admin1?.name}
          </LabelizedField>
        </Grid>
        <Grid item xs={3}>
          <LabelizedField label={t('Administrative Level 2')}>
            {ticket.household.admin2?.name}
          </LabelizedField>
        </Grid>
        <Grid item xs={3}>
          <LabelizedField label={t('Geolocation')}>
            {ticket.household.geopoint
              ? `${ticket.household.geopoint.coordinates[0]}, ${ticket.household.geopoint.coordinates[1]}`
              : '-'}
          </LabelizedField>
        </Grid>
        <Grid item xs={3}>
          <LabelizedField label={t('UNHCR CASE ID')}>
            {ticket.household?.unhcrId}
          </LabelizedField>
        </Grid>
      </Grid>
    </ApproveBox>
  );
};
