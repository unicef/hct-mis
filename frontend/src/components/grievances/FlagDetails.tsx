import {
  Box,
  Button,
  makeStyles,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Typography,
} from '@material-ui/core';
import moment from 'moment';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { DATE_FORMAT } from '../../config';
import { GRIEVANCE_TICKET_STATES } from '../../utils/constants';
import {
  GrievanceTicketDocument,
  GrievanceTicketNode,
  useApproveSystemFlaggingMutation,
} from '../../__generated__/graphql';
import { useConfirmation } from '../core/ConfirmationDialog';
import { FlagTooltip } from '../core/FlagTooltip';
import { Title } from '../core/Title';
import { UniversalMoment } from '../core/UniversalMoment';
import { ApproveBox } from './GrievancesApproveSection/ApproveSectionStyles';
import { ViewSanctionList } from './ViewSanctionList';

export function FlagDetails({
  ticket,
  canApproveFlag,
}: {
  ticket: GrievanceTicketNode;
  canApproveFlag: boolean;
}): React.ReactElement {
  const { t } = useTranslation();
  const confirm = useConfirmation();
  const useStyles = makeStyles(() => ({
    table: {
      minWidth: 100,
    },
  }));
  const [approve] = useApproveSystemFlaggingMutation({
    refetchQueries: () => [
      {
        query: GrievanceTicketDocument,
        variables: { id: ticket.id },
      },
    ],
  });
  const classes = useStyles();
  const confirmationText = t(
    'Are you sure you want to confirm flag (sanction list match) ?',
  );
  const removalText = t('Are you sure you want to remove the flag ?');
  const details = ticket.systemFlaggingTicketDetails;
  const isFlagConfirmed = details.approveStatus;
  return (
    <ApproveBox>
      <Title>
        <Box display='flex' justifyContent='space-between'>
          <Typography variant='h6'>{t('Flag Details')}</Typography>
          <Box>
            <ViewSanctionList
              referenceNumber={details.sanctionListIndividual.referenceNumber}
            />
            {canApproveFlag && (
              <Button
                disabled={
                  ticket.status !== GRIEVANCE_TICKET_STATES.FOR_APPROVAL
                }
                onClick={() =>
                  confirm({
                    content: isFlagConfirmed ? removalText : confirmationText,
                  }).then(() =>
                    approve({
                      variables: {
                        grievanceTicketId: ticket.id,
                        approveStatus: !details.approveStatus,
                      },
                    }),
                  )
                }
                variant='outlined'
                color='primary'
              >
                {isFlagConfirmed ? t('REMOVE FLAG') : t('CONFIRM FLAG')}
              </Button>
            )}
          </Box>
        </Box>
      </Title>
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell align='left' />
            <TableCell align='left'>{t('Ref. No. on Sanction List')}</TableCell>
            <TableCell align='left'>{t('Full Name')}</TableCell>
            <TableCell align='left'>{t('Date of Birth')}</TableCell>
            <TableCell align='left'>{t('National Ids')}</TableCell>
            <TableCell align='left'>{t('Source')}</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          <TableRow>
            <TableCell align='left'>
              {isFlagConfirmed ? (
                <FlagTooltip
                  message={t('Sanction List Confirmed Match')}
                  confirmed
                />
              ) : (
                ''
              )}
            </TableCell>
            <TableCell align='left'>-</TableCell>
            <TableCell align='left'>
              {details.goldenRecordsIndividual.fullName}
            </TableCell>
            <TableCell align='left'>
              <UniversalMoment>
                {details.goldenRecordsIndividual.birthDate}
              </UniversalMoment>
            </TableCell>
            <TableCell align='left'>
              {details.goldenRecordsIndividual.documents.edges
                .filter((item) => item.node.type.type === 'NATIONAL_ID')
                .map((item) => item.node.documentNumber)
                .join(', ') || '-'}
            </TableCell>
            <TableCell align='left'>{t('Golden Record')}</TableCell>
          </TableRow>
          <TableRow>
            <TableCell align='left' />
            <TableCell align='left'>
              {details.sanctionListIndividual.referenceNumber}
            </TableCell>
            <TableCell align='left'>
              {details.sanctionListIndividual.fullName}
            </TableCell>
            <TableCell align='left'>
              {details.sanctionListIndividual.datesOfBirth.edges
                .map((item) => moment(item.node.date).format(DATE_FORMAT))
                .join(', ') || '-'}
            </TableCell>
            <TableCell align='left'>
              {details.sanctionListIndividual.documents.edges
                .map((item) => item.node.documentNumber)
                .join(', ') || '-'}
            </TableCell>
            <TableCell align='left'>{t('Sanction List')}</TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </ApproveBox>
  );
}
