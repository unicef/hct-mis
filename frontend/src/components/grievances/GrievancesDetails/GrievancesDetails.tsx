import { Box, Grid, GridSize, Typography } from '@material-ui/core';
import React from 'react';
import { useTranslation } from 'react-i18next';
import {
  GRIEVANCE_CATEGORIES,
  GRIEVANCE_ISSUE_TYPES,
} from '../../../utils/constants';
import {
  grievanceTicketBadgeColors,
  choicesToDict,
  grievanceTicketStatusToColor,
  renderUserName,
} from '../../../utils/utils';
import {
  GrievancesChoiceDataQuery,
  GrievanceTicketQuery,
} from '../../../__generated__/graphql';
import { ContainerColumnWithBorder } from '../../core/ContainerColumnWithBorder';
import { ContentLink } from '../../core/ContentLink';
import { LabelizedField } from '../../core/LabelizedField';
import { OverviewContainer } from '../../core/OverviewContainer';
import { PhotoModal } from '../../core/PhotoModal/PhotoModal';
import { StatusBox } from '../../core/StatusBox';
import { Title } from '../../core/Title';
import { UniversalMoment } from '../../core/UniversalMoment';

interface GrievancesDetailsProps {
  ticket: GrievanceTicketQuery['grievanceTicket'];
  choicesData: GrievancesChoiceDataQuery;
  baseUrl: string;
  canViewHouseholdDetails: boolean;
  canViewIndividualDetails: boolean;
}

export const GrievancesDetails = ({
  ticket,
  choicesData,
  baseUrl,
  canViewHouseholdDetails,
  canViewIndividualDetails,
}: GrievancesDetailsProps): React.ReactElement => {
  const { t } = useTranslation();
  const statusChoices: {
    [id: number]: string;
  } = choicesToDict(choicesData.grievanceTicketStatusChoices);

  const priorityChoicesData = choicesData.grievanceTicketPriorityChoices;
  const urgencyChoicesData = choicesData.grievanceTicketUrgencyChoices;

  const categoryChoices: {
    [id: number]: string;
  } = choicesToDict(choicesData.grievanceTicketCategoryChoices);

  const issueType = ticket.issueType
    ? choicesData.grievanceTicketIssueTypeChoices
        .filter((el) => el.category === ticket.category.toString())[0]
        .subCategories.filter(
          (el) => el.value === ticket.issueType.toString(),
        )[0].name
    : '-';

  const showIssueType =
    ticket.category.toString() ===
      GRIEVANCE_CATEGORIES.SENSITIVE_GRIEVANCE.toString() ||
    ticket.category.toString() ===
      GRIEVANCE_CATEGORIES.DATA_CHANGE.toString() ||
    ticket.category.toString() ===
      GRIEVANCE_CATEGORIES.GRIEVANCE_COMPLAINT.toString();
  const showProgramme =
    ticket.issueType !== +GRIEVANCE_ISSUE_TYPES.ADD_INDIVIDUAL;
  const showPartner =
    ticket.issueType === +GRIEVANCE_ISSUE_TYPES.PARTNER_COMPLAINT;

  const mappedDocumentation = (): React.ReactElement => {
    return (
      <Box display='flex' flexDirection='column'>
        {ticket.documentation?.length
          ? ticket.documentation.map((doc) => {
              if (doc.contentType.includes('image')) {
                return (
                  <PhotoModal
                    key={doc.id}
                    src={doc.filePath}
                    variant='link'
                    linkText={doc.name}
                  />
                );
              }
              return (
                <ContentLink key={doc.id} download href={doc.filePath}>
                  {doc.name}
                </ContentLink>
              );
            })
          : '-'}
      </Box>
    );
  };

  const renderPaymentUrl = (): React.ReactElement => {
    if (ticket?.paymentRecord?.objType === 'PaymentRecord') {
      return (
        <ContentLink
          href={`/${baseUrl}/payment-records/${ticket.paymentRecord.id}`}
        >
          {ticket.paymentRecord.caId}
        </ContentLink>
      );
    }
    if (ticket?.paymentRecord?.objType === 'Payment') {
      return (
        <ContentLink
          href={`/${baseUrl}/payment-module/payments/${ticket.paymentRecord.id}`}
        >
          {ticket.paymentRecord.caId}
        </ContentLink>
      );
    }

    return <>-</>;
  };

  return (
    <Grid item xs={12}>
      <ContainerColumnWithBorder>
        <Title>
          <Typography variant='h6'>{t('Details')}</Typography>
        </Title>
        <OverviewContainer>
          <Grid container spacing={6}>
            {[
              {
                label: t('Status'),
                value: (
                  <StatusBox
                    status={statusChoices[ticket.status]}
                    statusToColor={grievanceTicketStatusToColor}
                  />
                ),
                size: 3,
              },
              {
                label: t('Priority'),
                value: (
                  <StatusBox
                    status={
                      priorityChoicesData[
                        priorityChoicesData.findIndex(
                          (obj) => obj.value === ticket.priority,
                        )
                      ]?.name || '-'
                    }
                    statusToColor={grievanceTicketBadgeColors}
                  />
                ),
                size: 3,
              },
              {
                label: t('Urgency'),
                value: (
                  <StatusBox
                    status={
                      urgencyChoicesData[
                        urgencyChoicesData.findIndex(
                          (obj) => obj.value === ticket.urgency,
                        )
                      ]?.name || '-'
                    }
                    statusToColor={grievanceTicketBadgeColors}
                  />
                ),
                size: 3,
              },
              {
                label: t('Assigned to'),
                value: renderUserName(ticket.assignedTo),
                size: 3,
              },
              {
                label: t('Category'),
                value: <span>{categoryChoices[ticket.category]}</span>,
                size: 3,
              },
              showIssueType && {
                label: t('Issue Type'),
                value: <span>{issueType}</span>,
                size: 3,
              },
              {
                label: t('Household ID'),
                value: (
                  <span>
                    {ticket.household?.id ? (
                      <ContentLink
                        href={
                          canViewHouseholdDetails
                            ? `/${baseUrl}/population/household/${ticket.household.id}`
                            : undefined
                        }
                      >
                        {ticket.household.unicefId}
                      </ContentLink>
                    ) : (
                      '-'
                    )}
                  </span>
                ),
                size: 3,
              },
              {
                label: t('Individual ID'),
                value: (
                  <span>
                    {ticket.individual?.id ? (
                      <ContentLink
                        href={
                          canViewIndividualDetails
                            ? `/${baseUrl}/population/individuals/${ticket.individual.id}`
                            : undefined
                        }
                      >
                        {ticket.individual.unicefId}
                      </ContentLink>
                    ) : (
                      '-'
                    )}
                  </span>
                ),
                size: showIssueType ? 3 : 6,
              },
              {
                label: t('Payment ID'),
                value: <span>{renderPaymentUrl()}</span>,
                size: showProgramme || showPartner ? 3 : 12,
              },
              showProgramme && {
                label: t('Programme'),
                value: ticket.programs?.name,
                size: showPartner ? 3 : 9,
              },
              showPartner && {
                label: t('Partner'),
                value: ticket.partner?.name,
                size: 6,
              },
              {
                label: t('Created By'),
                value: renderUserName(ticket.createdBy),
                size: 3,
              },
              {
                label: t('Date Created'),
                value: <UniversalMoment>{ticket.createdAt}</UniversalMoment>,
                size: 3,
              },
              {
                label: t('Last Modified Date'),
                value: <UniversalMoment>{ticket.updatedAt}</UniversalMoment>,
                size: 6,
              },
              {
                label: t('Administrative Level 2'),
                value: ticket.admin,
                size: 3,
              },
              {
                label: t('Area / Village / Pay point'),
                value: ticket.area,
                size: 3,
              },
              {
                label: t('Languages Spoken'),
                value: ticket.language,
                size: 3,
              },
              {
                label: t('Documentation'),
                value: mappedDocumentation(),
                size: 3,
              },
              {
                label: t('Description'),
                value: ticket.description,
                size: 12,
              },
              {
                label: t('Comments'),
                value: ticket.comments,
                size: 12,
              },
            ]
              .filter((el) => el)
              .map((el) => (
                <Grid key={el.label} item xs={el.size as GridSize}>
                  <LabelizedField label={el.label}>{el.value}</LabelizedField>
                </Grid>
              ))}
          </Grid>
        </OverviewContainer>
      </ContainerColumnWithBorder>
    </Grid>
  );
};
