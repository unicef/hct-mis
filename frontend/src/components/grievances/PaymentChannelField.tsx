import { Grid, IconButton } from '@material-ui/core';
import { Delete } from '@material-ui/icons';
import { Field } from 'formik';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { FormikTextField } from '../../shared/Formik/FormikTextField';
import { AllIndividualsQuery } from '../../__generated__/graphql';
import { LabelizedField } from '../core/LabelizedField';

export interface PaymentChannelProps {
  index: number;
  baseName: string;
  onDelete: () => {};
  isEdited?: boolean;
  paymentChannel?: AllIndividualsQuery['allIndividuals']['edges'][number]['node']['paymentChannels'][number];
}

export function PaymentChannelField({
  index,
  baseName,
  onDelete,
  isEdited,
  paymentChannel,
}: PaymentChannelProps): React.ReactElement {
  const { t } = useTranslation();
  return (
    <>
      <Grid item xs={11} />
      {!isEdited ? (
        <Grid item xs={1}>
          <IconButton onClick={onDelete}>
            <Delete />
          </IconButton>
        </Grid>
      ) : null}
      <Grid item xs={4}>
        <LabelizedField label={t('Payment channel item')} value='IBAN' />
      </Grid>
      <Grid item xs={4}>
        <LabelizedField
          label={t('Current Value')}
          value={paymentChannel?.bankAccountNumber}
        />
      </Grid>
      <Grid item xs={3}>
        <Field
          name={`${baseName}[${index}].bankAccountNumber`}
          fullWidth
          variant='outlined'
          label={t('New Value')}
          component={FormikTextField}
          required
        />
      </Grid>
      <Grid item xs={4}>
        <LabelizedField
          label={t('Payment channel item')}
          value={t('Bank name')}
        />
      </Grid>
      <Grid item xs={4}>
        <LabelizedField
          label={t('Current Value')}
          value={paymentChannel?.bankName}
        />
      </Grid>
      <Grid item xs={3}>
        <Field
          name={`${baseName}[${index}].bankName`}
          fullWidth
          variant='outlined'
          label={t('New Value')}
          component={FormikTextField}
          required
        />
      </Grid>
    </>
  );
}
