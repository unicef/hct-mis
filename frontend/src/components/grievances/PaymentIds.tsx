import { Box, Typography } from '@material-ui/core';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { PaymentVerificationNode, PaymentRecordNode } from '../../__generated__/graphql';
import { ContentLink } from '../core/ContentLink';
import { Title } from '../core/Title';
import { ApproveBox } from './GrievancesApproveSection/ApproveSectionStyles';

type VerificationId = {
  id: PaymentVerificationNode["id"];
  caId: PaymentRecordNode["caId"];
}

interface PaymentIdsProps {
  verifications: VerificationId[];
}

export const PaymentIds = ({ verifications }: PaymentIdsProps): React.ReactElement => {
  const { t } = useTranslation();
  const businessArea = useBusinessArea();

  const mappedIds = verifications.map(
    (verification): React.ReactElement => (
      <Box mb={1}>
        <ContentLink
          href={`/${businessArea}/verification-records/${verification.id}`}
        >
          {verification.caId}
        </ContentLink>
      </Box>
    ),
  );
  return (
    <ApproveBox>
      <Title>
        <Typography variant='h6'>{t('Payment Ids')}</Typography>
      </Title>
      <Box display='flex' flexDirection='column'>
        {mappedIds}
      </Box>
    </ApproveBox>
  );
};
