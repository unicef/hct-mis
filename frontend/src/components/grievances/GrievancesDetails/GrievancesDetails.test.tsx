import React from 'react';
import { render } from '../../../testUtils/testUtils';
import { fakeGrievanceTicket } from '../../../../fixtures/grievances/fakeGrievanceTicket';
import { fakeGrievancesChoices } from '../../../../fixtures/grievances/fakeGrievancesChoices';
import { GrievancesDetails } from './GrievancesDetails';

describe('components/grievances/GrievancesDetails', () => {
  it('should render', () => {
    const { container } = render(
      <GrievancesDetails
        businessArea='afghanistan'
        ticket={fakeGrievanceTicket}
        choicesData={fakeGrievancesChoices}
        canViewHouseholdDetails
        canViewIndividualDetails
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
