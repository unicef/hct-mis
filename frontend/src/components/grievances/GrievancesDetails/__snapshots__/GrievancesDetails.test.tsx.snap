// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/grievances/GrievancesDetails should render 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiPaper-root sc-bdVaJa fUBeDk MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="sc-gzVnrw AoKEu"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Details
        </h6>
      </div>
      <div
        class="sc-ifAKCX iZCSws"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                STATUS
              </span>
              <div>
                <div
                  class="sc-EHOje gFjtOh"
                >
                  <div
                    class="sc-bZQynM kQLhqL"
                    data-cy="status-container"
                  >
                    In Progress
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                CATEGORY
              </span>
              <div>
                <span>
                  Data Change
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                Issue Type
              </span>
              <div>
                <span>
                  Withdraw Individual
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                HOUSEHOLD ID
              </span>
              <div>
                <span>
                  <a
                    class="sc-bwzfXH emFVa"
                    href="/afghanistan/population/household/SG91c2Vob2xkTm9kZTo3NjExNzM2Ny0yYWFiLTRmNTEtODUwOC1mMzBmODliYWUzYzE="
                  >
                    HH-20-0000.0001
                  </a>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                INDIVIDUAL ID
              </span>
              <div>
                <span>
                  <a
                    class="sc-bwzfXH emFVa"
                    href="/afghanistan/population/individuals/SW5kaXZpZHVhbE5vZGU6ZjVmMzgwN2ItYTBjOS00Mjk0LTg0Y2QtNDhkMjllMjZhODU3"
                  >
                    IND-88-0000.0005
                  </a>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                PAYMENT ID
              </span>
              <div>
                <span>
                  -
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                CONSENT
              </span>
              <div>
                <span>
                  Yes
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                CREATED BY
              </span>
              <div>
                <span>
                  Maciej Szewczyk
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                DATE CREATED
              </span>
              <div>
                <time
                  datetime="1644571534682"
                >
                  11 Feb 2022
                </time>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                LAST MODIFIED DATE
              </span>
              <div>
                <time
                  datetime="1644571964102"
                >
                  11 Feb 2022
                </time>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                DESCRIPTION
              </span>
              <div>
                <span>
                  dwdw
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                ASSIGNED TO
              </span>
              <div>
                <span>
                  Maciej Szewczyk
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                ADMINISTRATIVE LEVEL 2
              </span>
              <div>
                <span />
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                AREA / VILLAGE / PAY POINT
              </span>
              <div>
                <span />
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div>
              <span
                class="sc-htpNat hcDmtB"
                color="textSecondary"
              >
                LANGUAGES SPOKEN
              </span>
              <div>
                <span>
                  -
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
